utils/normalize-securecookie.js

```diff
@@ -0,0 +1,151 @@
+'use strict'
+
+const crypto = require('crypto')
+
+const fs = require('fs')
+
+const path = require('path')
+
+const regexes = [
+  {
+    from: /<securecookie\s+(host|name)=\s*["']\^?\.[+*]?\$?["']\s+(host|name)=\s*["']\^?\.[+*]?\$?["']\s*\/>/g,
+    to: '<securecookie host=".+" name=".+" />'
+  },
+  {
+    from: /<securecookie\s+host=\s*["'](.+)["']\s+name=\s*["']\^?\.[+*]?\$?["']\s*\/>/g,
+    to: '<securecookie host="$1" name=".+" />'
+  },
+  {
+    from: /<securecookie\s+name=\s*["']\^?\.[+*]?\$?["']\s+host=\s*["'](.+)["']\s*\/>/g,
+    to: '<securecookie host="$1" name=".+" />'
+  },
+  {
+    from: /<securecookie\s+host=\s*["']\^?\.[+*]?\$?["']\s+name=\s*["'](.+)["']\s*\/>/g,
+    to: '<securecookie host=".+" name="$1" />'
+  },
+  {
+    from: /<securecookie\s+name=\s*["'](.+)["']\s+host=\s*["']\^?\.[+*]?\$?["']\s*\/>/g,
+    to: '<securecookie host=".+" name="$1" />'
+  }
+]
+
+function parseWhitelist (data) {
+  const lines = data.split('\n').slice(1)
+  const result = {}
+
+  for (const line of lines) {
+    const lineSplit = line.split(',')
+
+    if (lineSplit.length !== 4) continue
+
+    const key = lineSplit[3].toLowerCase()
+
+    const value = lineSplit[0]
+
+    result[key] = value
+  }
+
+  return result
+}
+
+function readFile (path) {
+  return new Promise((resolve, reject) => {
+    fs.readFile(path, 'utf8', (err, data) => {
+      if (err) {
+        reject(err)
+      } else {
+        resolve(data)
+      }
+    })
+  })
+}
+
+function writeFile (path, data) {
+  return new Promise((resolve, reject) => {
+    fs.writeFile(path, data, 'utf8', (err) => {
+      if (err) {
+        reject(err)
+      } else {
+        resolve()
+      }
+    })
+  })
+}
+
+function readdir (path) {
```

Comment:
Just a nit / hint: if you're targetting only the latest Node versions (which I think makes sense anyway for scripts, especially given that we use ES6), you can just do `const readdir = util.promisify(fs.readdir);`. (`util.promisify` is a recently added built-in exactly for wrapping Node.js APIs into Promise ones).

Reply:
@RReverser Done.